content/manager/broker-security.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: bt_wiki
+title: Broker Security (RabbitMQ)
+category: Manager
+draft: false
+weight: 1000
+
+---
+
+Cloudify uses RabbitMQ as its broker, and supports configurable security.
+
+# Authentication
+
+When bootstrapping, the cloudify manager must be provided with credentials for rabbit. These will use default values if not overridden in the inputs.
+
+## Username
+
+It is suggested that you change the username to something other than the default. It is recommended that you use only upper and lower case letters and numbers for the username.
+
+The username can be set using the rabbitmq_username input to the 3.3 manager blueprint.
+
+## Password
+
+It is recommended that you set the password to something strong that is known only to those who are authorized.
+
+It is recommended that the password is comprised of only ASCII characters, excluding quotes, line feeds, and dollar signs, as these may cause the password to be set incorrectly during the bootstrap.
+
+The password should be set using the rabbitmq_password input to the 3.3 manager blueprint.
+
+## With external broker
+
+If you are using an external broker you must have correctly configured the user on the external RabbitMQ broker. This user must have full permissions on the root (/) vhost.
+
+No changes of configuration of the external broker will be performed.
+
+# Network Encryption
+
+Communications with the broker can be secured (with some exceptions, see below) using SSL/TLS.
+
+A certificate from a certificate authority is not required, as the provided public certs will be 'pinned' (they will be the only certificate valid for connections to the broker).
+
+Currently you will likely need to create the manager with a specified IP in order to appropriately create the certificate.
+
+## Generating a certificate
+
```

Comment:
It means that when bootstrapping, one needs to put the manager on a static IP, though a DNS name (or one provided on all nodes through /etc/hosts) would also work assuming our agent installer would use the name rather than the IP.
Can we actually do that? I just realised I'm not entirely sure.

I think the only way around this would be to generate the cert on the manager during bootstrap (as it would then have its IP). On the bright side, that'd actually be a more secure way to do it (in the sense that the key would never leave the machine it was supposed to live on).